Makefile

```diff
@@ -59,7 +59,7 @@ man-pages: $(HELP_ALL:=.ronn) $(HELP_ALL) $(HELP_ALL:=.txt)
 	groff -Wall -mtty-char -mandoc -Tutf8 -rLL=$(TEXT_WIDTH)n $< | col -b >$@
 
 %.1: %.1.ronn bin/ronn
-	bin/ronn --organization=GITHUB --manual="Hub Manual" share/man/man1/*.ronn
+	bin/ronn --organization=GITHUB --manual="Hub Manual" $<
```

Comment:
Hi thank you for trying to help out with this! But I don't get this optimization. Before, this patch, this command reports taking under 4.5s:

```sh
git clean -fdx share; time make man-pages
```

After your patch, it takes more than 12s.

So, I'm seeing 3x *slowdown* instead of 3x speedup.

Reply:
I was testing in openSUSE with hub-2.3.0-pre10 

There we do `make bin/hub man-pages`
I think, that might make a difference, because git only contains one .ronn file as input, but I saw more. Those extras are generated in some step.

`strace` shows hub-remote.1.ronn being created by
```bash
["/bin/sh", "-c", "bin/hub help hub-remote --plain-text | script/format-ronn hub-remote share/man/man1/hub-remote.1.ronn"]
["script/format-ronn", "hub-remote", "share/man/man1/hub-remote.1.ronn"]
```

I'm wondering though, where your slowdown comes from, because it should be exactly the same when you just have 1 .ronn file from git as input.
Can you reproduce the slowdown?
Do you get 24 entries under `/usr/share/man/man1/hub*.1*` in both cases?

Reply:
Maybe your measurement/cleanup did not account for the one-time creation of `bin/hub` ?

Reply:
I've ensured that `make bin/hub` is already run before measurement. Your optimization should apply only to the `man-pages` step, so including `bin/hub` build step in the measurement doesn't make sense.

What do you get with the following script on your branch?

```
make bin/hub
git clean -fdxq share; time make man-pages >/dev/null 2>&1
git checkout HEAD^ -- Makefile
git clean -fdxq share; time make man-pages >/dev/null 2>&1
```

Reply:
I had to simulate it, because I cannot get it to build from git
```
> make
script/build -o bin/hub
main.go:8:2: cannot find package "github.com/github/hub/commands" in any of:
        /usr/lib64/go/1.10/src/github.com/github/hub/commands (from $GOROOT)
        /home/abuild/go/src/github.com/github/hub/commands (from $GOPATH)
        /usr/share/go/1.10/contrib/src/github.com/github/hub/commands
main.go:9:2: cannot find package "github.com/github/hub/github" in any of:
```
`git bisect run bash -c "make clean ; make` 
says, build was broken by commit 00e1b9957d4093fa675ba6a213dfa2cf71186819

I noticed that we sed bin/ronn to ronn, using the system's ruby2.5-rubygem-ronn-0.7.3 but OTOH 0.7.3 is also the latest.

```bash
cp -a Makefile.unpatched Makefile ; rm -f share/man/man1/* ; cp -a orig/* share/man/man1/ ; time make man-pages >/dev/null 2>&1

real    0m13.063s
user    0m13.020s
sys     0m1.235s

cp -a Makefile.patched Makefile ; rm -f share/man/man1/* ; cp -a orig/* share/man/man1/ ; time make man-pages >/dev/null 2>&1
        
real    0m4.131s
user    0m3.987s
sys     0m0.451s
```

Using a local revert of 00e1b9957d4093fa675ba6a213dfa2cf71186819, I even got your lines to work.

```
git checkout make
git clean -fdxq share; time make man-pages >/dev/null 2>&1

real    0m4.329s
user    0m4.173s
sys     0m0.477s
git checkout master
git clean -fdxq share; time make man-pages >/dev/null 2>&1

real    0m1.230s
user    0m1.323s
sys     0m0.227s
```

There seems to also be a downside content-wise in rendering each page indiviually:
```diff
-<p><a class="man-ref" href="hub-am.1.html">hub-am<span class="s">(1)</span></a>, <a class="man-ref" href="hub.1.html">hub<span class="s">(1)</span></a>, <span class="man-ref">git-apply<span class="s">(1)</span></span></p>
+<p><span class="man-ref">hub-am<span class="s">(1)</span></span>, <span class="man-ref">hub<span class="s">(1)</span></span>, <span class="man-ref">git-apply<span class="s">(1)</span></span></p>
```


but then I found that parallelism plays a role there:
```
git checkout master
git clean -fdxq ; mkdir -p bin; ln -sf /usr/bin/ronn bin/ronn ; time make -j2 man-pages >/dev/null 2>&1 ; ls share/man/man1/|wc -l

real    0m8.037s
user    0m15.778s
sys     0m1.555s
100

> git checkout make
Switched to branch 'make'
> git clean -fdxq ; mkdir -p bin; ln -sf /usr/bin/ronn bin/ronn ; time make -j2 man-pages >/dev/null 2>&1 ; ls share/man/man1/|wc -l

real    0m2.633s
user    0m4.913s
sys     0m0.578s
100
```